up-l1/someip.adoc

```diff
@@ -0,0 +1,265 @@
+
+
+== Application Layer Translation
+
+Application (or message payload) translation is the process of converting *SOME/IP-SD* subscription and discovery messages, to/from *uDiscovery* and *uSubscription* Messages.
+
+=== uSubscription
+
+The following section will elaborate only on the translation of *uSubscription* messages to/from *SOME/IP-SD* messages. Subscription state (persistent or not) is handled in the *uSubscription* services and not at the transport layer or this component.
+
+The following section we will elaborate on how Eventgroup Entry types are mapped to link:../up-l3/usubscription/v3/README.adoc[*uSubscription*] messages for the subscribe
+and unsubscribe flows per `[PRS_SOMEIPSD_00385]`.
+
+==== Common Fields
+
+<<common-field-mappings>> table below illustrates the common *SOME/IP-SD* EventGroup Entry fields that are present in for all *SOME/IP-SD* Eventgroup entry types (`SubscribeEventgroup`, `SubscribeEventGroupAck`, `SubscribeEventgroupNack`, `StopSubscribeEventGroup`).
+
+These fields are then mapped to uProtocol `UUri` attributes used in uProtocol `UMessage` for performing subscription operations.
+
+.Common Field Mappings
+[#common-field-mappings,  cols="1,2"]
+|===
+| Eventgroup Entry Field | UUri
+
+a| `Service ID` a| Set in lower 16 bits of `ue_id`
+
+a| `Instance ID` a| If instance is not the default (`0x1`), set it in upper 16 bits of `ue_id`
+
+a| `Major Version` a| `ue_version_major`
+
+a| `EventGroup ID` / `Event ID` a| `resource_id`
+
+NOTE: `1:1` mapping between `EventGroup ID` and `Event ID` is assumed. This may require specific ECU Firmware.
+
+
+|===
+
+NOTE: `UUri.authority_name` *MAY* be translated to/from IPv4 (and/or IPv6) Endpoint Option of the *SOME/IP-SD* message, although in `vsomeip` this is not available in the API (e.g. each discovered Endpoint maps to `Service ID`/`Instance ID`/`Major Version`/`Minor Version`).
+
+==== EventGroup Entry Type Mapping
+
+.EventGroup Entry Type Mapping
+[#eventgroup-entry-mapping, cols="1,1,2"]
+|===
+| Eventgroup Entry Type | uSubscription Message | Additional Details
+
+a| `SubscribeEventGroup`
+a| `SubscriptionRequest`
+a| The message is used to subscribe to a topic.
+
+* If `SubscribeAttributes.expire` is not set, `TTL` *MAY* be set to `0xFFFFFF` to indicate that the subscription should remain for the duration of the ignition cycle
+
+NOTE: `vsomeip` has static TTL configuration (for Service Discovery), that applies for all subscriptions. It can't be changed per subscription.
+
+a| `SubscribeEventGroupAck`
+a| `SubscriptionResponse`
+a| The message is used to acknowledge a successful subscription request.
+
+* `SubscriptionStatus.code` *SHALL* be set to `OK`
+* `SubscriptionStatus.state` *SHALL* be set to `SUBSCRIBED`
+
+a| `SubscribeEventGroupNack`
+a| `SubscriptionResponse`
+a| The message is used to acknowledge a failed subscription request.
+
+* SubscriptionStatus.code *SHALL* be set to the corresponding error code per the <<commstatus-mapping>> table
```

Comment:
This is changing in https://github.com/eclipse-uprotocol/up-spec/pull/163, it might be better to just state that ,the error codes below will be returned and not through SubscriptionStatus.code (it will be in commstatus of UAttributes)